### PR TITLE
feat(TPSVC-6043): collapsible panels virtualized list

### PR DESCRIPTION
**Pull request by andrii-holovko-dev**
**What is the problem this PR is trying to solve?**
Create common component for collapsible panels virtualized list

**What is the chosen solution to this problem?**
Use VirtualizedList to render CollapsiblePanels

**Please check if the PR fulfills these requirements**

* [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md)

### CODE REVIEW

File: packages/components/src/VirtualizedList/RowCollapsiblePanel/RowCollapsiblePanel.component.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+		let onRowDoubleClick;
+
+		if (parent.props.onRowDoubleClick) {
+			onRowDoubleClick = event => parent.props.onRowDoubleClick({ event, rowData });
+		}
+
+		const onToggle = (event, measure) => {
+			if (parent.props.onRowClick) {
+				parent.props.onRowClick({ event, rowData: { ...rowData, index } });
+				setTimeout(measure, 0);
```
Comment (jsomsanith) [inline packages/components/src/VirtualizedList/RowCollapsiblePanel/RowCollapsiblePanel.component.js]: I don't understand the defer here.
But that doesn't work if you already have expanded panels as initial props.
You can instead use onLoad attribute on the div to execute measure()

File: packages/components/src/VirtualizedList/RowCollapsiblePanel/RowCollapsiblePanel.component.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+				columnIndex={0}
+				key={index}
+				parent={parent}
+				rowIndex={index}
+			>
+				{({ measure }) => (
+					<div
+						className={classNames(css['tc-collapsible-row'], rowData.className, className)}
+						id={id}
+						onDoubleClick={onRowDoubleClick}
```
Comment (jsomsanith) [inline packages/components/src/VirtualizedList/RowCollapsiblePanel/RowCollapsiblePanel.component.js]: Why do you set a double clic gesture ?

File: packages/components/src/VirtualizedList/RowCollapsiblePanel/RowCollapsiblePanel.component.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+						onKeyDown={e => onKeyDown(e, this.ref)}
+						ref={ref => {
+							this.ref = ref;
+						}}
+						role="listitem"
+						tabIndex="0"
+						aria-posinset={index + 1}
+						aria-setsize={parent.props.rowCount}
+						aria-label={titleField && getCellData(titleField, parent, index)}
```
Comment (jsomsanith) [inline packages/components/src/VirtualizedList/RowCollapsiblePanel/RowCollapsiblePanel.component.js]: This is always undefined. You don't pass any column definition so no title column, so the `extractSpecialFields` doesn't work here.
You need to use the panel info to get the title.

File: packages/components/src/VirtualizedList/RowCollapsiblePanel/RowCollapsiblePanel.component.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+						style={{ ...style, maxHeight: rowData.expanded ? 'none' : `${DEFAULT_HEIGHT}px` }}
```
Comment (jsomsanith) [inline packages/components/src/VirtualizedList/RowCollapsiblePanel/RowCollapsiblePanel.component.js]: Why `maxHeight ` is redefined here ? Shouldn't you let CellMeasurer do the workd to set the height ?

File: packages/components/src/VirtualizedList/RowCollapsiblePanel/__snapshots__/RowCollapsiblePanel.test.js.snap [hunk 5]
```diff
@@ -0,0 +1,91 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`RowCollapsiblePanel should render collapsible panel row 1`] = `
+<VirtualizedList(RowCollapsiblePanel)
```
Comment (jsomsanith) [inline packages/components/src/VirtualizedList/RowCollapsiblePanel/__snapshots__/RowCollapsiblePanel.test.js.snap]: Your snapshot isn't meaningful. It prints your component with its props, but that doesn't test your component.

File: packages/components/src/VirtualizedList/RendererSelector.component.js [hunk 6]
```diff
@@ -94,6 +99,14 @@ class RendererSelector extends React.Component {
 		} else {
 			ListRenderer = ListGrid;
 			customProps = { rowRenderer: getRowRenderer(type, this.props.rowRenderers) };
+
+			if (type === COLLAPSIBLE_PANEL) {
```
Comment (jsomsanith) [inline packages/components/src/VirtualizedList/RendererSelector.component.js]: Please avoid specific code in the RendererSeletor. If you need to pass options, perhaps you can :
- consider that the rowRenderer can hold an attribute `options`
- so 
```
ListRenderer = ListGrid;
const rowRenderer = getRowRenderer(type, this.props.rowRenderers);
customProps = { rowRenderer, ...rowRenderer.options };
```
- in RowCollapsiblePanel/index.js
```
const cache = new CellMeasurerCache({ fixedWidth: true });
RowCollapsiblePanel.options = {
	deferredMeasurementCache: cache,
	rowHeight: cache.rowHeight,
};
export default RowCollapsiblePanel;
```

File: packages/components/src/index.js [hunk 7]
```diff
@@ -155,7 +155,6 @@ export {
 	HttpError,
 	Icon,
 	IconsProvider,
-	Image,
```
[the remark this answers is not in the document]
Comment (andrii-holovko-dev) [reply packages/components/src/index.js]: Image is not imported here

File: packages/components/src/CollapsiblePanel/CollapsiblePanel.scss [hunk 8]
```diff
@@ -271,3 +271,7 @@ $tc-toggle-color: #555964 !default;
 		}
 	}
 }
+
+.panel-list {
```
[the remark this answers is not in the document]
Comment (andrii-holovko-dev) [reply packages/components/src/CollapsiblePanel/CollapsiblePanel.scss]: I pass this class name as "theme" property of CollapsiblePanel. It is used to make CollapsiblePanel inside list fit the guideline

File: packages/components/stories/VirtualizedList.js [hunk 9]
```diff
@@ -710,4 +774,41 @@ storiesOf('Virtualized List', module)
 				</VirtualizedList>
 			</section>
 		</div>
-	));
+	))
+	.add('List > Debug Virtualized Collapsible list', () => {
```
Comment (jmfrancois) [inline packages/components/stories/VirtualizedList.js]: please remove this example, seems you have added to debug.

File: packages/components/src/CollapsiblePanel/CollapsiblePanel.scss (hunk 8 above)
[the remark this answers is not in the document]
Comment (jmfrancois) [reply packages/components/src/CollapsiblePanel/CollapsiblePanel.scss]: This is not the way to fit the guideline.

the RowCollapsiblePanel component already customize the CollapsiblePanel.

So please move the margin-bottom there.

File: packages/components/src/VirtualizedList/RowCollapsiblePanel/RowCollapsiblePanel.component.js [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+				cache={options.deferredMeasurementCache}
+				columnIndex={0}
+				key={this.props.index}
+				parent={this.props.parent}
+				rowIndex={index}
+			>
+				{({ measure }) => (
+					// eslint-disable-next-line jsx-a11y/no-static-element-interactions
+					<div
+						className={classNames(css['tc-collapsible-row'], rowData.className, className)}
```
Comment (jmfrancois) [inline packages/components/src/VirtualizedList/RowCollapsiblePanel/RowCollapsiblePanel.component.js]: So I would add it here (the global class) for customization: `tc-collapsible-row`

File: packages/components/src/VirtualizedList/RowCollapsiblePanel/RowCollapsiblePanel.scss [hunk 11]
```diff
@@ -0,0 +1,9 @@
+.tc-collapsible-row {
+	:global(.panel-collapse) {
+		transition-duration: 0s;
+	}
```
Comment (jmfrancois) [inline packages/components/src/VirtualizedList/RowCollapsiblePanel/RowCollapsiblePanel.scss]: You can add your margin 10 px here.:
```
:global(.panel) {
  margin-bottom: $padding-small;
}
```

**Pull request merged**